src/notevennexus/validators.py

```diff
@@ -0,0 +1,269 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+import numpy as np
+
+from .tree import Dataset, Group
+from .validate import Validator, Violation
+
+
+class NX_class_attr_missing(Validator):
+    def __init__(self) -> None:
+        super().__init__("NX_class_attr_missing", "NX_class attribute is missing")
+
+    def applies_to(self, node: Dataset | Group) -> bool:
+        return isinstance(node, Group)
+
+    def validate(self, node: Dataset | Group) -> Violation | None:
+        if "NX_class" not in node.attrs:
+            return Violation(node.name)
+
+
+class depends_on_target_missing(Validator):
```

Comment:
```suggestion
class Depends_on_target_missing(Validator):
```

Reply:
Also use uppercase for all the other classes in this file?

Reply:
The naming matches the things it refers to: It is called `depends_on` in NeXus. I think it is less confusing this way? Otherwise you could also argue class names should be CamelCase, and this would become `DependsOnTargetMissing` or `Depends_onTargetMissing` or something weird.